OMEdit/OMEditGUI/Animation/AbstractAnimationWindow.cpp

```diff
@@ -207,9 +214,131 @@ void AbstractAnimationWindow::createActions()
   mpRotateCameraRightAction = new QAction(QIcon(":/Resources/icons/rotateCameraRight.svg"), tr("Rotate Right"), this);
   mpRotateCameraRightAction->setStatusTip(tr("Rotates the camera right"));
   connect(mpRotateCameraRightAction, SIGNAL(triggered()), this, SLOT(rotateCameraRight()));
+  //interactive control action
+  mpInteractiveControlAction = mpAnimationParameterDockerWidget->toggleViewAction();
+  mpInteractiveControlAction->setIcon(QIcon(":/Resources/icons/control-panel.svg"));
+  mpInteractiveControlAction->setText(tr("interactive control"));
+  mpInteractiveControlAction->setStatusTip(tr("Opens the interactive control panel"));
+  connect(mpInteractiveControlAction, SIGNAL(triggered()), this, SLOT(initInteractiveControlPanel()));
+  mpAnimationParameterDockerWidget->hide();
+}
+
+/*!
+ * \brief AbstractAnimationWindow::updateControlPanelValues
+ */
+void AbstractAnimationWindow::updateControlPanelValues()
+{
+  if (getVisualizer()) {
+    VisualizerFMU* FMUvis = dynamic_cast<VisualizerFMU*>(mpVisualizer);
+    for (unsigned int stateIdx = 0; stateIdx < mSpinBoxVector.size(); stateIdx++)
+        {
+          mStateLabels.at(stateIdx)->setText(QString::number(FMUvis->getFMU()->getFMUData()->_states[stateIdx]));
+        }
+  }
+}
+
+/*!
+ * \brief AbstractAnimationWindow::initInteractiveControlPanel
+ */
+void AbstractAnimationWindow::initInteractiveControlPanel()
+{
+  if (getVisualizer()) {
+    VisualizerFMU* FMUvis = dynamic_cast<VisualizerFMU*>(mpVisualizer);
+    QWidget *widget = new QWidget(this);
+    mSpinBoxVector.clear();
+    mStateLabels.clear();
+
+    if (FMUvis->getFMU()->getFMUData()->_stateNames.size()==FMUvis->getFMU()->getFMUData()->_nStates)
+    {
```

Comment:
I suggest putting the curly braces in the same line like you did in the above if condition.